node/src/components/consensus/utils.rs

```diff
@@ -52,5 +58,170 @@ pub(crate) fn check_sufficient_finality_signatures(
         });
     }
 
+    // Verify: the total weight decreased by the smallest weight of a single signature should be
+    // below threshold (anti-spam countermeasure).
+    if weight_minus_minimum * U512::from(*lower_bound.denom())
+        > total_weight * U512::from(*lower_bound.numer())
+    {
+        return Err(FinalitySignatureError::TooManySignatures {
+            trusted_validator_weights: trusted_validator_weights.clone(),
+            block_signatures: Box::new(block_signatures.clone()),
+            signature_weight: Box::new(signature_weight),
+            weight_minus_minimum: Box::new(weight_minus_minimum),
+            total_validator_weight: Box::new(total_weight),
+            finality_threshold_fraction,
+        });
+    }
+
     Ok(())
 }
+
+pub(crate) fn get_minimal_set_of_signatures(
+    trusted_validator_weights: &BTreeMap<PublicKey, U512>,
+    finality_threshold_fraction: Ratio<u64>,
+    mut block_signatures: BlockSignatures,
+) -> Option<BlockSignatures> {
+    // Calculate the values for comparison.
+    let total_weight: U512 = trusted_validator_weights
+        .iter()
+        .map(|(_, weight)| *weight)
+        .sum();
+
+    let lower_bound = (finality_threshold_fraction + 1) / 2;
```

Comment:
So generally speaking we err on the paranoid side and demand ≥2/3rds of the weight.  This is the most we can demand given what we assume about liveness.

However, from the security assumptions we _could_ require just >1/3rd of the weight.  This is because we assume that at most 1/3rd of the weight is byzantine.  If we had more than 1/3rd of the weight we would know we have at least one honest node.

Do we want this function to support the space of permissible `lower_bound` parameters, which is (1/3,2/3]? 

WDYAT? CC @afck

Reply:
I agree… actually, what are we currently using in other places? We should just be consistent.

Reply:
We seem to be using 1/2 here, too: https://github.com/casper-network/casper-node/blob/98fc4b0362af500e370200ab5ee0b6860c999e88/node/src/components/consensus/utils.rs#L40

Either way, we should deduplicate this. And I'm happy to switch to 2/3 instead!

Reply:
Wait - with FTT being 1/3, this _is_ 2/3, isn't it? (1/3 + 1)/2 = 2/3.

I think Matt is suggesting switching to >1/3 instead for the threshold for finality signatures, ie. > FTT - as that still means at least 1 signature was from an honest validator.

Or to > x, where x is between 1/3 and 2/3.

Reply:
> I think Matt was suggesting to make this configurable, too?

Yes, I am suggesting this should be configurable.

The lowest level of security is demanding > 1/3rd = FTT.  In this case, we know we have one honest validator.

The highest level of security is demanding > 2/3rd = (FTT + 1) / 2.  We can't demand higher than this because we allow for 1/3rd of the stake to be down.

Reply:
We decided that we'll go with `(ftt + 1)/2` for now.